### PR TITLE
CB-11963 Handle recipe execution failure on hosts without metadata or…

### PR DESCRIPTION
**Pull request by Bajzathd**
… fqdn

See detailed description in the commit message.

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/recipe/OrchestratorRecipeExecutor.java [hunk 1]
```diff
@@ -176,30 +177,42 @@ private String getRecipeTimeoutErrorMessage(CloudbreakOrchestratorTimeoutExcepti
...
-        }).collect(Collectors.joining("\n ---------------------------------------------- \n")
-        );
+        String message = failures.stream()
+                .map(failure -> getSingleRecipeExecutionFailureMessage(instanceMetaData, failure))
+                .collect(Collectors.joining("\n ---------------------------------------------- \n"));
         return new StringBuilder("Failed to execute recipe(s): \n").append(message).toString();
     }
 
+    @VisibleForTesting
+    String getSingleRecipeExecutionFailureMessage(Set<InstanceMetaData> instanceMetaData, RecipeFailure failure) {
```
Comment (horadla23) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/recipe/OrchestratorRecipeExecutor.java:195]: I would prefer not using this `@VisibleForTesting` but unit test it through public method and make this private

**Review by topolyai5 — COMMENTED** [review]
What will the result if the host and fqdn are null as well?

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/recipe/OrchestratorRecipeExecutor.java (hunk 1 above)
Comment (Bajzathd) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/recipe/OrchestratorRecipeExecutor.java:195]: This is called from another private method, that handles error from 4 different public methods, this is why I have done it this way. To eliminate this annotation a better solution would be to extract this logic to its own class (or into RecipeExecutionFailureCollector) and test it instead, I can do it if you think that would be better.

**Comment by Bajzathd** [comment]
> What will the result if the host and fqdn are null as well?

If both values are null than they would equal, but the host can not be null, it is what we get back from the recipe execution failure.

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/recipe/OrchestratorRecipeExecutor.java (hunk 1 above)
Comment (pkedvessy) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/recipe/OrchestratorRecipeExecutor.java:195]: +1 on reducing the complexity of this class by moving out the failure message generation logic into a separated component

Comment (horadla23) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/recipe/OrchestratorRecipeExecutor.java:195]: i think testing only one public method's error handling would be enough to test this method

**Review by horadla23 — APPROVED** [review]

**Pull request merged**